pandas/core/dtypes/cast.py

```diff
@@ -965,12 +965,16 @@ def maybe_cast_to_datetime(value, dtype, errors='raise'):
                         if is_datetime64:
                             value = to_datetime(value, errors=errors)._values
                         elif is_datetime64tz:
-                            # input has to be UTC at this point, so just
-                            # localize
-                            value = (to_datetime(value, errors=errors)
-                                     .tz_localize('UTC')
-                                     .tz_convert(dtype.tz)
-                                     )
+                            is_dt_string = is_string_dtype(value)
+                            value = to_datetime(value, errors=errors)
+                            if is_dt_string:
```

Comment:
hmm adding this incremental logic is error prone. I think this should simply be converted higher up. can you see if this can be simplied that way.

Reply:
This logic is pretty high up the stack already.

The Series constructor calls [`_sanitize_array`](https://github.com/pandas-dev/pandas/blob/83436af8ae1ccad49b7ceac7471c060d823d10ab/pandas/core/series.py#L257) which then calls [`maybe_cast_to_datetime`](https://github.com/pandas-dev/pandas/blob/83436af8ae1ccad49b7ceac7471c060d823d10ab/pandas/core/series.py#L2985) (which I am editing here) fairly early. The line I am editing is the first instance where arrays of numbers (and strings) are converted with `to_datetime`

Reply:
try passing utc=True to to_datetime

this logic is too specific here

Reply:
actually it might be that

to_datetime(...., tz=dtype.tz) might be the correct idiom here (another PR implements this)

Reply:
Yeah #17413 will make this easier.

However I think there will need to be logic to determine if the incoming data is naive or already localized to UTC. The code comments that were already here note that (numeric) data is already UTC while string data (what I am trying to fix here) may be naive.

Reply:
my point is solving this here is wrong 

this is a bug in to_datetime itself

Reply:
passing utc=True to to_datetime will make this work (then just converting to the dtype)

Reply:
can you try passing ``utc=True`` here

Reply:
`utc=True` does not work here because the timestamp string should be localized to the specified dtype tz directly intead of UTC in between.

```
# with utc=True
In [5]: ts_naive = '2013-01-01 00:00:00'
   ...:  result = Series([ts_naive], dtype='datetime64[ns, CET]')
   ...:  expected = Series([pd.Timestamp(ts_naive, tz='CET')])

In [6]: result
Out[6]: 
0   2013-01-01 01:00:00+01:00
dtype: datetime64[ns, CET]

In [7]: expected
Out[7]: 
0   2013-01-01 00:00:00+01:00
dtype: datetime64[ns, CET]
```